### PR TITLE
Modification to spawn templating approach to set zone nominal loads programmatically 

### PR DESCRIPTION
**Pull request by amyeallen1**
This PR addresses issue 400 ([https://github.com/urbanopt/geojson-modelica-translator/issues/400](url), by modifying the system design parameters schema to contain arrays of zone nominal heating and cooling loads for a Spawn model, which are used to set the nominal load values in the template. The test input files have also been updated accordingly.

**Assumptions** 
This approach assumes that nominal loads will be provided in the same order in which the zone names are listed. 

The scaling factor is no longer applied to the nominal loads themselves, so they should be set accounting for the scaling factor, and have been in the test input files. 

Nick @nllong  or Ted @macintoshpie , does one of you want to review this? Thanks!

### CODE REVIEW

File: geojson_modelica_translator/system_parameters/schema.json [hunk 1]
```diff
@@ -427,6 +427,20 @@
           "items": {
             "type": "string"
           }
+        },
+        "zone_nom_htg_loads": {
```
Comment (macintoshpie) [inline geojson_modelica_translator/system_parameters/schema.json:431]: Should we add these new parameters to the "required" list below? e.g. what happens if users don't provide this?
Another option is to create a default somehow.

**Review by macintoshpie — CHANGES_REQUESTED** [review]
I can't comment too much on the modelica/modeling correctness, but I left some feedback for python things. Thanks!

File: geojson_modelica_translator/model_connectors/load_connectors/spawn.py [hunk 2]
```diff
@@ -134,6 +140,8 @@ def to_modelica(self, scaffold, keep_original_models=False):
                 "temp_setpoint_heating": temp_setpoint_heating,
                 "temp_setpoint_cooling": temp_setpoint_cooling,
             },
+            "zone_nom_htg_loads": str(repr(zone_nom_htg_loads)).replace("[", "{").replace("]", "}").split("rray(", 1)[-1],
```
[the remark this answers is not in the document]
Comment (amyeallen1) [reply geojson_modelica_translator/model_connectors/load_connectors/spawn.py:144]: Good point--I added a comment.

File: geojson_modelica_translator/model_connectors/load_connectors/teaser.py [hunk 3]
```diff
@@ -558,7 +558,8 @@ def post_process(self, scaffold, keep_original_models=False):
                 ) + 273.15
             }
         }
-
+        print("nominal heat flow")
+        print(building_template_data['nominal_heat_flow'])
```
[the remark this answers is not in the document]
Comment (amyeallen1) [reply geojson_modelica_translator/model_connectors/load_connectors/teaser.py]: Done!

File: geojson_modelica_translator/system_parameters/schema.json [hunk 4]
```diff
@@ -427,6 +427,20 @@
...
+          "description": "Array of zone nominal heating loads (values should be >0) ",
+          "type": "array",
+          "items": {
+            "type": "integer"
```
[the remark this answers is not in the document]
Comment (amyeallen1) [reply geojson_modelica_translator/system_parameters/schema.json]: Good point--I changed them both to "number."

File: geojson_modelica_translator/system_parameters/schema.json (hunk 1 above)
Comment (amyeallen1) [reply geojson_modelica_translator/system_parameters/schema.json:431]: Good point--I added them both to the "required" list.

**Comment by amyeallen1** [comment]
Thanks, Ted, for reviewing, and for the useful comments! I made your suggested revisions. 

On further thought, I also realized that it would be better to maintain the use of the scaling factor in the Spawn template as it was before, so I restored that, and changed the test input values accordingly. This screenshot verifies the expected outcome in Modelica (for the mixed loads test as an example), that is, that the nominal load values in each zone are equal to the input values divided by the scaling factor (which has a value of 5) 

image

**Review by macintoshpie — APPROVED** [review]
Looks good from the python side of things! @nllong leaning on you for modelica/modeling review

**Comment by amyeallen1** [comment]
Thanks, Ted! @macintoshpie @nllong Nick, I think this is pretty straight forward from the Modelica perspective--the screen shot in the comment above shows the nominal loads set as expected for the test scenario.

**Review by nllong — APPROVED** [review]
baller! only future recommendation would be to name as `zone_nominal_heating_loads` since there is no reason to abbreviate everything.

**Comment by amyeallen1** [comment]
Thanks, Nick! Roger that! @nllong

**Pull request merged**